crates/core/tedge_api/src/errors.rs

```diff
@@ -0,0 +1,11 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum PluginError {
+    #[error("The sender could not transfer the message to its receiving end. Did it get closed?")]
+    CouldNotSendMessage(#[from] tokio::sync::mpsc::error::SendError<crate::Message>),
+    #[error("An error in the configuration was found")]
+    Configuration(#[from] toml::de::Error),
```

Comment:
An error in the configuration is not necessarily a `toml::de::Error`! I would like to propose to rename this variant to `ConfigurationDeserialization` and add a new variant `Configuration` that represents the "error in the configuration" idea. 

We might need to add some subvariants or simply make it a `anyhow::Error`, so that plugins can specify configuration errors nicely...